tests/python/relay/test_external_codegen.py

```diff
@@ -0,0 +1,223 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Unit tests for graph partitioning."""
+import os
+import numpy as np
+import pytest
+
+import tvm
+import tvm.relay.testing
+import tvm.relay.transform
+from tvm import relay
+from tvm.contrib import util
+
+def check_result(mod, map_inputs, out_shape, result, tol=1e-5):
+    with relay.build_config(opt_level=3, disabled_pass=["AlterOpLayout"]):
+        json, lib, _ = relay.build(mod, "llvm")
+    test_dir = os.path.dirname(os.path.realpath(os.path.expanduser(__file__)))
+    source_dir = os.path.join(test_dir, "..", "..", "..")
+    contrib_path = os.path.join(source_dir, "src", "runtime", "contrib")
+
+    kwargs = {}
+    kwargs["options"] = ["-O2", "-std=c++11", "-I" + contrib_path]
+    tmp_path = util.tempdir()
+    lib_name = 'lib.so'
```

Comment:
This doesn't work on Windows, maybe better to skip testing if on windows.

Reply:
I think this should be okay on Windows as we just reused the current API. export_library should have been used on Windows, right? But I do not have a Windows machine to verify.

Reply:
Actually export_library has issues on Windows and it doesn't work out of the box if I remember correctly. For this case, MSVC doesn't understand the same command line args as gcc, and lib name should be lib.dll

Reply:
ahh, you are right. Thanks for pointing out.